pandas/_libs/dtypes.pxd

```diff
@@ -5,13 +5,40 @@ Common location for shared fused types
 from numpy cimport (
     float32_t,
     float64_t,
+    int8_t,
+    int16_t,
+    int32_t,
     int64_t,
+    uint8_t,
+    uint16_t,
+    uint32_t,
     uint64_t,
 )
 
+ctypedef fused numeric_t:
+    int8_t
+    int16_t
+    int32_t
+    int64_t
+
+    uint8_t
+    uint16_t
+    uint32_t
+    uint64_t
+
+    float32_t
+    float64_t
+
 ctypedef fused numeric_object_t:
+    numeric_t
+    object
+
+ctypedef fused iu_64_floating_t:
```

Comment:
numeric_64_t ?

Reply:
Tricky part here is that this is `i64 + u64 + f32 + f64` so not only 64-bit types. Used `floating` to refer to using all float types because that's the name for the `cython` builtin which includes `f32` and `f64`

Reply:
yeah what you have is prob fine as you explained; if we have a better name in the future can change.